### PR TITLE
Bug 1812730 - Prevent collections with empty name from being created

### PR DESCRIPTION
**Pull request by gitstart**
### What
The PR will disable the dialog `OK` button and keyboard `DONE` button if the collection name is `blank`

### Method 1

1. Open a tab with a web page open
2. Click on `three dot menu`
3. Select `Save to collection`
4. Delete the default name
5. Try saving by clicking on the keyboard `Done` button
6. Repeat steps `1 - 3`  
7. Enter a name with spaces at the beginning and end 

### Method 2
1. Make sure you have more than 1 tab open with a web page open
2. Click on the tab tray
3. Click on `three dot menu`
4. Click on `Select tabs`
5. Select one or more tabs
6. Click on `Collection button`
7. Click on ~Add new collection`
8. Delete the name and try saving or
9. Append and prepend spaces on the name 

### Screenshots
| Issue video | Fix Video |
| ----------- | --------- |
| https://www.loom.com/share/6880e832f260420ab3a26a5a88045420 | https://www.loom.com/share/5155001bee58479c91856252c611affe|

### Pull Request checklist
 - [X] Quality: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
 - [ ] Tests: This PR includes thorough tests or an explanation of why it does not
 - [ ] Changelog: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
 - [ ] Accessibility: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features
### After merge
- Milestone: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- Breaking Changes: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

---
This code was written and reviewed by GitStart Community. Growing future engineers, one PR at a time.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1812730

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/collections/CollectionsDialog.kt [hunk 1]
```diff
@@ -128,8 +132,29 @@ internal fun CollectionsDialog.showAddNewDialog(
...
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                dialog.getButton(DialogInterface.BUTTON_POSITIVE).isClickable = s.toString().isNotBlank()
+            }
+
+            override fun afterTextChanged(s: Editable?) {
+                /* noop */
+            }
+        },
+    )
```
Comment (titooan) [inline fenix/app/src/main/java/org/mozilla/fenix/collections/CollectionsDialog.kt:150]: You can use the [TextView.doOnTextChanged](https://android.googlesource.com/platform/frameworks/support/+/refs/heads/androidx-main/core/core-ktx/src/main/java/androidx/core/widget/TextView.kt#42) extension function to make the code cleaner. 

```suggestion
    collectionNameEditText.doOnTextChanged { text, _, _, _ ->
        dialog.getButton(DialogInterface.BUTTON_POSITIVE).isClickable = text.toString().isNotBlank()
    }
```

**Review by titooan — CHANGES_REQUESTED** [review]
Great job, looks good to me! 

I only suggested a small code improvment, and after the code is rebased it shoud be ready to merge.

Thanks for this contribution!

**Comment by gitstart** [comment]
Hi @titooan Thanks for reviewing PR. Requested changes has been pushed. Can you please re-review?

**Comment by titooan** [comment]
Sorry for the delay, and thanks for applying the changes! It looks good to me.

**Comment by titooan** [comment]
bors try

**Comment by titooan** [comment]
bors try

**Review by titooan — APPROVED** [review]

**Pull request merged**